packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -387,10 +387,10 @@ export class MediaPlaceholder extends Component {
 
 	render() {
 		const {
-			dropZoneUIOnly,
+			showMediaSelectionUI,
```

Comment:
@gziolo 👋 would changing the prop name here be a problem? Are there some steps we need to take before changing it?

Reply:
It'd be a breaking change as this prop is part of the public API.

The corresponding PR was merged in March (#12367) and WordPress 5.2 went out on May 7th (https://wordpress.org/news/2019/05/jaco/). I don't see it in `wp/5.2` branch so it means it was merged during the beta phase and not included in the release cycle.

Anyway, it's been a few months since it was exposed as part of the public API so we would have to keep the old name as well to ensure we don't break any plugins which might be using it.

Reply:
I see it documented in the `master` branch but not here:

https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/media-placeholder#dropzoneuionly

> If true, only the Drop Zone will be rendered. No UI controls to upload the media will be shown. The `disableDropZone` prop still takes precedence over `dropZoneUIOnly` – specifying both as true will result in nothing to be rendered.

@jorgefilipecosta - can you help here?

Reply:
So, as far as I understand we are not able to change that prop name. 
Should we then share the `dropZoneUIOnly ` prop (which sounds weird a bit on mobile) between platforms?

Reply:
When I look at the latest [docs](https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/media-placeholder#dropzoneuionly) I see that disableDropZone and dropZoneUIOnly are also related: `The disableDropZone prop still takes precedence over dropZoneUIOnly – specifying both as true will result in nothing to be rendered` Thus, just renaming it as `showMediaSelectionUI` won't make sense unless we refactor it to work independently.
AFAIU, we'll either need to change the behavior of web implementation too or we'll keep the props same but we'll need to add to docs that drag & drop is not yet supported on mobile. I am aiming towards to the latter one since that's a nice feature to have on mobile too. wdyt @koke ?

Reply:
Those look a bit confusing to me, and I'd prefer to flip them both 😅 

- `showMediaSelectionUI` (`== ! dropZoneUIOnly`)
- `showDropZone` (`== ! disableDropZone`)

It's still possible to set them both as `false` and have it render nothing, but I think it's clearer that that would happen from these names, and the props wouldn't seem to contradict each other.

> So, as far as I understand we are not able to change that prop name

What I understand is that we still need to support the old props. We can leave only the new ones in the documentation and all the uses across the codebase, but the implementation should still look for the old props and behave accordingly, in case a plugin is using those.

Reply:
> that's a nice feature to have on mobile too(in the future)

Definitely agree. While it's not yet on the roadmap, we should support drag and drop media upload on mobile at some point

Reply:
As far as I understand, disabling the drop zone is a very rare case. So I find the `disableDropZone` style configuration more suitable in this case, because if we convert it to `showDropZone` we'll either need to:
- set this to true in almost all places that uses this component 
or 
- give a default true inside the component which wouldn't be as intuitive as the current usage

On the other hand I agree with the `showMediaSelectionUI` change but this time I'd expect this value wouldn't be overridden by `disableDropZone`. Because the new name conveys that it should be responsible for just showing/hiding the media selection UI, it is not conveying any info about the relation with the drop zone. I understand that making these 2 independent is just the thing we want but I really don't know how we can do that transition in a smooth way. In the deprecation phase we'll need to support both new and old ones together, but `showMediaSelectionUI` will be acting differently so it brings a certain complexity to the logic.

So if we are going to add drag&drop in the following 3 months for example, does it worth the effort? That's the thing I am trying to figure. I mean it is up to us to plan for this.

Reply:
> give a default true inside the component which wouldn't be as intuitive as the current usage

I don't understand why that wouldn't be intuitive. I'm fine with leaving the name, since we don't really need to change it for this, as all we need is what you said here:

> I understand that making these 2 independent is just the thing we want

I think we're fine on that front? If we have `showMediaSelectionUI` and `disableDropZone`, those would be independent:

showMediaSelectionUI | disableDropZone | renders
---------------------|-----------------|--------
false                | false           | drop zone
false                | true            | nothing
true                 | false           | media selection
true                 | true            | drop zone + media selection

I think this is the same behavior as before, but previously the name `dropZoneUIOnly` would seem to contradict `disableDropZone` when both were true.

Reply:
I thought currently we'd see nothing if we set:

````
disableDropZone={ true } 
dropZoneUIOnly={ false }
```

because I was under the impression that disableDropZone will disable all the things related to dropZone since it has precedence. But it looks like it is not the case, I did a quick experiment in my local:

<img width="916" alt="Screen Shot 2019-09-05 at 19 57 14" src="https://user-images.githubusercontent.com/5032900/64363254-cedea100-d018-11e9-8e6a-be1a6ae39bbd.png">

So basically they do act independently in this sense, just like how we wanted. This makes transition phase simple because we won't need to deal with changing the logic. What is more, I think new name makes more sense with the current behavior imho. But ofc we need to get this reviewed.

Reply:
So let's try to list some steps to propose such a change properly for a review:

1. Add showMediaSelectionUI
2. Deprecate `dropZoneUIOnly`
3. Use these together for a while, like:

before:

``` 
 if ( dropZoneUIOnly )
``` 

after:

```
 if ( dropZoneUIOnly &&  ! showMediaSelectionUI )
``` 

Anything else I am missing?

BTW I've never deprecated a prop from components before but I guess it looks like [this](https://github.com/WordPress/gutenberg/blob/a491a79ab2ec9c05b296c50e1ce0c5e389e19175/packages/block-editor/src/components/rich-text/index.js#L297)

Reply:
Another thing worth noting might be dropZoneUIOnly is representing almost the MediaPlaceholder itself, I mean if I set it to true I see nothing:

<img width="708" alt="Screen Shot 2019-09-05 at 20 55 29" src="https://user-images.githubusercontent.com/5032900/64367627-30a30900-d021-11e9-844d-fe585fab1348.png">

So again I am not sure about using a positive keyword that has a default `true` value and will be true in almost all the cases. It feels more natural to me to use sth like `hidePlaceholder` with a default `false`. Because when I see showXXX I question myself like do I need to set this to true explicitly and I feel the need to check the docs. On the other hand with a name like `hidePlaceholder` or `hideMediaSelectionUI` I am sure at first sight that it won't get hidden until I set it to `true`. That was the reason I said it was more intuitive. Not a big deal but just wanted to mention.

Reply:
That makes sense. For consistency with the other prop I would suggest disableMediaSelection instead of “hide”

Reply:
`disableMediaSelection` sounds good to me 👍

Reply:
@jorgefilipecosta let us know if you have concerns about this. Otherwise I think we can go ahead and deprecate `dropZoneUIOnly` and add `disableMediaSelection`?

Reply:
@pinarol 	I've updated, my PR according to the conversation above:

* added `disableMediaSelection` prop
* deprecate`dropZoneUIOnly` prop